src/ast/postprocessing/structured_documentation_reconstruction.ts

```diff
@@ -81,12 +114,22 @@ export class StructuredDocumentationReconstructor {
 
         let stopOnNextGap = false;
 
+        const rxRemoveGarbage = /^[^/]+/;
```

Comment:
Is this rx meant to remove leading spaces before the `///` or `*` ?

Reply:
Not only the spaces, but everything before the first `/`. This is because the cases with dangling nodes:
```solidity
{
    /// dangling
}
```
The AST does not specifically give any information about where the internal whitespace starts, so we **have to** look from very begining of `Block`'s source range, that includes `{\n\s` before `/`. This is even more uglier when we have ```solidity
unchecked {
    /** dangling */
}
```
or
```solidity
contract {
    /// dangling
}
```
Due to this, we have to remove any possible garbage before first leading `/` and then check if it is `///` or `/**`.

Reply:
I will document it better.

Reply:
Addressed in https://github.com/ConsenSys/solc-typed-ast/pull/115/commits/872be86fefb3d356d2552929b0a260684ffd6801